### PR TITLE
Connected tabular generator with dataset_generator

### PR DESCRIPTION
**Pull request by drahc1R**
Connected tabular generator with dataset_generator

### CODE REVIEW

File: synthetic_data/generators.py [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 from synthetic_data.graph_synthetic_data import GraphDataGenerator
 from synthetic_data.synthetic_data import make_data_from_report
 
 
 class TabularGenerator(BaseGenerator):
     """Class for generating synthetic tabular data."""
 
-    def __init__(self, profile, seed=None, noise_level: float = 0.0):
+    def __init__(
+        self, profile, seed=None, noise_level: float = 0.0, method: str = "correlated"
```
Comment (taylorfturner) [inline synthetic_data/generators.py:17]: wonder if `method` should be a boolean instead of a str option... like we could do `correlated: bool = True` and then limit to two option of correlated and simple...

File: tests/test_generators.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import os
+import unittest
+
+pass
```
Comment (taylorfturner) [inline tests/test_generators.py:4]: remove

File: tests/test_tabular_generator_integration
Comment (taylorfturner) [inline tests/test_tabular_generator_integration:1]: this needs a file extension

File: tests/test_tabular_generator_integration [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    #         if data_type == "datetime":
+    #             col_data.append({"data_type":data_type, "ordered":ordered, "date_format_list": "%m/%d/%Y, %H:%M:%S"})
+    #         else:
+    #             col_data.append({"data_type":data_type, "ordered":ordered})
+
+    #     return dg.generate_dataset_by_class(
+    #         rng=rng,
+    #         columns_to_generate=col_data,
+    #     )
```
Comment (taylorfturner) [inline tests/test_tabular_generator_integration:80]: why all commented out?

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: tests/test_tabular_generator_integration (hunk 3 above)
Comment (drahc1R) [reply tests/test_tabular_generator_integration]: WIP

File: synthetic_data/distinct_generators/int_generator.py [hunk 4]
```diff
@@ -20,4 +20,6 @@ def random_integers(
 
     :return: np array of integers
     """
+    if max_value <= 0:
```
Comment (drahc1R) [inline synthetic_data/distinct_generators/int_generator.py:23]: There is this edge case where the max_value in the report is 0. You cannot set 0 as the max_value in `rng.integers`, so I set this to the default max_value

File: synthetic_data/generators.py [hunk 5]
```diff
@@ -47,20 +53,116 @@ def post_profile_processing_w_data(cls, data, profile):
...
+            columns = self.profile.report()["data_stats"]
+            col_data = []
+
+            for col in columns:
+                generator = col.get("data_type", None)
+                order = col.get("order", None)
+                col_stats = col["statistics"]
+                min_value = col_stats.get("min", None)
+                max_value = col_stats.get("max", None)
+
```
Comment (drahc1R) [inline synthetic_data/generators.py:88]: Made a custom hashmap for each generator. Retrieved values from the report.

File: synthetic_data/generators.py [hunk 6]
```diff
@@ -47,20 +53,116 @@ def post_profile_processing_w_data(cls, data, profile):
...
+                    )
+
+                elif generator == "float":
+                    col_data.append(
+                        {
+                            "generator": generator,
+                            "name": "flo",
+                            "min_value": min_value,
+                            "max_value": max_value,
+                            "sig_figs": int(
```
Comment (drahc1R) [inline synthetic_data/generators.py:125]: casted float to int as needed

File: synthetic_data/generators.py [hunk 7]
```diff
@@ -47,20 +53,116 @@ def post_profile_processing_w_data(cls, data, profile):
...
+                                "name": "txt",
+                                "chars": col_stats.get("vocab", None),
+                                "str_len_min": min_value,
+                                "str_len_max": max_value,
+                                "order": order,
+                            },
+                        )
+
+            return generate_dataset(
+                rng=rng, columns_to_generate=col_data, dataset_length=num_samples
```
Comment (drahc1R) [inline synthetic_data/generators.py:164]: assuming num_samples is the dataset_length

**Comment by drahc1R** [comment]
Will add tests tomorrow

**Comment by taylorfturner** [comment]
> Will add tests tomorrow

got it. Thanks!

File: synthetic_data/distinct_generators/text_generator.py [hunk 8]
```diff
@@ -39,6 +39,10 @@ def random_text(
         )
     text_list = []
 
+    # edge case
+    if str_len_min == str_len_max:
+        str_len_max += 1
+
```
Comment (drahc1R) [inline synthetic_data/distinct_generators/text_generator.py:45]: Another edge case where str_len_max == str_len_min in report. Doesn't work in rng.integers, so add one to to str_len_max as it's non-inclusive in function

File: synthetic_data/generators.py [hunk 9]
```diff
@@ -47,20 +57,114 @@ def post_profile_processing_w_data(cls, data, profile):
...
+                order = col.get("order", None)
+                col_stats = col["statistics"]
+                min_value = col_stats.get("min", None)
+                max_value = col_stats.get("max", None)
+
+                if generator == "datetime":
+                    date_format = col_stats["format"]
+                    start_date = pd.to_datetime(
+                        col_stats.get("min", None), format=date_format[0]
+                    )
```
Comment (drahc1R) [inline synthetic_data/generators.py:94]: Realized here that "format" in report returns a List[str] and not just a str

File: tests/test_generators.py [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+            options=self.profile_options,
+            samples_per_update=len(self.tab_data),
+        )
+        instance = TabularGenerator(profile)
+        instance.method = "correlated"
+        instance.synthesize(num_samples=10)
+        mock_make_data.assert_called_once()
+
+    @mock.patch("synthetic_data.generators.generate_dataset")
+    def test_uncorrelated_synthesize_columns_to_generate(self, mock_generate_dataset):
```
Comment (drahc1R) [inline tests/test_generators.py:81]: This test checks if the expected columns_to_generate hashmap was generated correctly. Checks every item to the expected and checks if the actual hashmap is a subset of expected

File: tests/test_generators.py [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+        actual_synthetic_data = generator.synthesize(20)
+
+        self.assertEqual(len(actual_synthetic_data), 20)
+        self.assertIsInstance(actual_synthetic_data, pd.DataFrame)
+        np.testing.assert_array_equal(
+            actual_synthetic_data.columns.values,
+            np.array(["dat", "txt", "int", "flo"], dtype="object"),
+        )
+
+        # for column in actual_synthetic_data_columns:
```
Comment (drahc1R) [inline tests/test_generators.py:274]: will delete all of this in a second, but essentially was trying to hardcode an expected, but obviously the data is generated randomly, so we can only check for general stuff: length, type, and column values of output

File: synthetic_data/synthetic_data.py [hunk 12]
```diff
@@ -402,6 +402,7 @@ def make_data_from_report(
     n_samples: int = None,
     noise_level: float = 0.0,
     seed=None,
+    is_correlated: bool = True,
```
Comment (taylorfturner) [inline synthetic_data/synthetic_data.py:405]: dont' need this

File: synthetic_data/synthetic_data.py [hunk 13]
```diff
@@ -429,7 +430,9 @@ def make_data_from_report(
     n_informative = len(report["data_stats"])
 
     # build covariance matrix
-    R = report["global_stats"]["correlation_matrix"]
+    R = np.eye(n_informative)
+    if is_correlated:
```
Comment (taylorfturner) [inline synthetic_data/synthetic_data.py:434]: dont' need this

File: synthetic_data/synthetic_data.py [hunk 14]
```diff
@@ -429,7 +430,9 @@ def make_data_from_report(
...
+        R = report["global_stats"]["correlation_matrix"]
```
Comment (taylorfturner) [inline synthetic_data/synthetic_data.py:435]: revert

File: tests/test_generators.py [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+        cls.uncorrelated_generator = Generator(
+            profile=cls.tab_profile, is_correlated=False, seed=42
+        )
+
+    def test_synthesize_tabular(self):
+        generator = Generator(profile=self.tab_profile, seed=42)
+        self.assertIsInstance(generator, TabularGenerator)
+        synthetic_data = generator.synthesize(100)
+        self.assertEqual(len(synthetic_data), 100)
+
```
Comment (taylorfturner) [inline tests/test_generators.py:46]: this looks dupes

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/distinct_generators/int_generator.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (taylorfturner) [reply synthetic_data/distinct_generators/int_generator.py:23]: or really... it needs to validate that the `min` / `max` parameters to `.integers` aren't equal

File: synthetic_data/distinct_generators/int_generator.py [hunk 16]
```diff
@@ -20,4 +20,7 @@ def random_integers(
 
     :return: np array of integers
     """
+    if min_value == max_value:
+        max_value += 1
```
Comment (drahc1R) [inline synthetic_data/distinct_generators/int_generator.py:24]: standard fix for min_value == max_value

File: synthetic_data/generators.py [hunk 17]
```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
...
+        else:
+            self.generate_column_data()
+
+            return generate_dataset(
+                rng=self.rng,
+                columns_to_generate=self.col_data,
+                dataset_length=num_samples,
+            )
+
+    def generate_column_data(self):
```
Comment (drahc1R) [inline synthetic_data/generators.py:87]: Refactored as a separate function in order to clean up synthesize()

File: synthetic_data/generators.py [hunk 18]
```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
...
-        )
+        if self.is_correlated:
+            return make_data_from_report(
+                report=self.profile.report(),
+                n_samples=num_samples,
+                noise_level=noise_level,
+                seed=self.tabular_generator_seed,
+            )
+        else:
+            self.generate_column_data()
```
Comment (drahc1R) [inline synthetic_data/generators.py:79]: call generate column data

File: synthetic_data/generators.py [hunk 19]
```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
...
+                        "max_value": max_value,
+                        "sig_figs": int(
+                            col_stats.get("precision", None).get("max", None)
+                        ),
+                        "order": order,
+                    }
+                )
+
+            elif generator == "string":
+                if col.get("categorical", False):
```
Comment (drahc1R) [inline synthetic_data/generators.py:142]: fixed issue of getting categorical in wrong part of report

File: tests/test_generators.py [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+        )
+        correlated_tabular_generator = TabularGenerator(profile)
+        self.assertTrue(correlated_tabular_generator.is_correlated)
+        correlated_tabular_generator.synthesize(num_samples=10)
+        mock_make_data.assert_called_once()
+
+    @mock.patch("synthetic_data.generators.generate_dataset")
+    def test_uncorrelated_synthesize_columns_to_generate(self, mock_generate_dataset):
+
+        generator = TabularGenerator(profile=self.profile, is_correlated=False, seed=42)
```
Comment (drahc1R) [inline tests/test_generators.py:58]: made dryer by only profiling data once in setup

File: tests/test_generators.py [hunk 21]
```diff
@@ -0,0 +1,234 @@
...
+        self.assertFalse(generator.is_correlated)
+        expected_columns_to_generate = [
```
Comment (drahc1R) [inline tests/test_generators.py:60]: fixed expected columns to generate with the correct expected output

File: tests/test_generators.py [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+                "name": "int",
+                "min_value": 664.0,
+                "max_value": 9464.0,
+                "order": "random",
+            },
+        ]
+        generator.synthesize(20)
+
+        mock_generate_dataset.assert_called_once()
+        for i in range(len(expected_columns_to_generate)):
```
Comment (drahc1R) [inline tests/test_generators.py:210]: testing each entry of expected with actual and then checking if the lists are equal by checking if actual is a subset of expected.

File: tests/test_generators.py [hunk 23]
```diff
@@ -0,0 +1,234 @@
...
+                        set(generator.col_data[i][key]).issubset(
+                            expected_columns_to_generate[i][key]
+                        )
+                    )
+                else:
+                    self.assertEqual(
+                        generator.col_data[i][key], expected_columns_to_generate[i][key]
+                    )
+
+    def test_uncorrelated_synthesize_output(self):
```
Comment (drahc1R) [inline tests/test_generators.py:223]: tests for general output constraints

File: synthetic_data/distinct_generators/int_generator.py (hunk 16 above)
Comment (taylorfturner) [reply synthetic_data/distinct_generators/int_generator.py:24]: if they are the same, though, why not just generate all the same values equal to len(num_rows)... in this scenario with `+=1`, you'll be generating data that is not equal to the original distribution

File: synthetic_data/distinct_generators/text_generator.py [hunk 24]
```diff
@@ -39,6 +39,10 @@ def random_text(
         )
     text_list = []
 
+    # edge case
+    if str_len_min == str_len_max:
+        str_len_max += 1
```
Comment (taylorfturner) [inline synthetic_data/distinct_generators/text_generator.py:44]: same as above

File: synthetic_data/generators.py (hunk 18 above)
Comment (taylorfturner) [reply synthetic_data/generators.py:79]: do you think there is a better naming here? maybe `generate_uncorrelated_data()`?

Comment (taylorfturner) [reply synthetic_data/generators.py:79]: `generate_column_data()` currently generates data from a report too ...

File: tests/test_generators.py [hunk 25]
```diff
@@ -0,0 +1,234 @@
...
+                "correlation.is_enabled": True,
+                "multiprocess.is_enabled": False,
+            }
+        )
+        dp.set_seed(0)
+
+        # create dataset and profile for tabular
+        cls.tab_data = dp.Data(os.path.join(test_dir, "data/iris.csv"))
+        cls.tab_profile = dp.Profiler(
+            cls.tab_data, profiler_type="structured", options=cls.profile_options
```
Comment (taylorfturner) [inline tests/test_generators.py:31]: do we need both?

File: synthetic_data/generators.py (hunk 17 above)
Comment (taylorfturner) [reply synthetic_data/generators.py:87]: I think we may be able to improve this still

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/distinct_generators/int_generator.py (hunk 16 above)
Comment (drahc1R) [reply synthetic_data/distinct_generators/int_generator.py:24]: i think it will generate values equal to the ones in the original
it is max index exclusive

File: synthetic_data/generators.py [hunk 26]
```diff
@@ -1,20 +1,32 @@
...
+    ):
         """Initialize tabular generator object."""
         super().__init__(profile, seed)
         self.noise_level = noise_level
+        self.is_correlated = is_correlated
+        if not seed:
+            self.tabular_generator_seed = self.seed
+        else:
+            self.tabular_generator_seed = seed
+        self.rng = np.random.default_rng(seed=self.tabular_generator_seed)
```
[the remark this answers is not in the document]
Comment (drahc1R) [reply synthetic_data/generators.py:28]: changing it to `f not seed:
            self.tabular_generator_seed = self.seed
        else:
            self.tabular_generator_seed = seed
        self.rng = np.random.default_rng(seed=self.tabular_generator_seed)`


this wouldn't work in synthesize() as it needs a way to get the seed somehow...

File: tests/test_generators.py [hunk 27]
```diff
@@ -0,0 +1,234 @@
...
+    def test_synthesize_correlated_method(self, mock_make_data):
+        profile = dp.Profiler(
+            data=self.tab_data,
+            options=self.profile_options,
+            samples_per_update=len(self.tab_data),
+        )
+        correlated_tabular_generator = TabularGenerator(profile)
+        self.assertTrue(correlated_tabular_generator.is_correlated)
+        correlated_tabular_generator.synthesize(num_samples=10)
+        mock_make_data.assert_called_once()
```
[the remark this answers is not in the document]
Comment (drahc1R) [reply tests/test_generators.py:53]: this isn't tested anywhere else

File: tests/test_generators.py [hunk 28]
```diff
@@ -0,0 +1,234 @@
...
+                    "groucho-oregon",
+                    "groucho-singapore",
+                    "groucho-tokyo",
+                    "groucho-sa",
+                    "zeppo-norcal",
+                    "groucho-norcal",
+                    "groucho-us-east",
+                    "groucho-eu",
+                ],
+                "probabilities": [
```
[the remark this answers is not in the document]
Comment (drahc1R) [reply tests/test_generators.py:82]: yes

File: synthetic_data/distinct_generators/int_generator.py (hunk 16 above)
Comment (taylorfturner) [reply synthetic_data/distinct_generators/int_generator.py:24]: add test for this in `distinct_generators/test_int_generator.py`

File: synthetic_data/distinct_generators/text_generator.py (hunk 24 above)
Comment (taylorfturner) [reply synthetic_data/distinct_generators/text_generator.py:44]: same here -- add test

File: synthetic_data/generators.py [hunk 29]
```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
...
-        )
+        if self.is_correlated:
+            return make_data_from_report(
+                report=self.profile.report(),
+                n_samples=num_samples,
+                noise_level=noise_level,
+                seed=self.tabular_generator_seed,
+            )
+        else:
+            self.generate_uncorrelated_column_data()
```
Comment (taylorfturner) [inline synthetic_data/generators.py:79]: assign to a variable `column_to_gen` and not a class attribute

File: synthetic_data/generators.py [hunk 30]
```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
...
+
+            return generate_dataset(
+                rng=self.rng,
+                columns_to_generate=self.col_data,
+                dataset_length=num_samples,
+            )
+
+    def generate_uncorrelated_column_data(self):
```
Comment (taylorfturner) [inline synthetic_data/generators.py:87]: this is just huge ... I think this could be improved

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: tests/distinct_generators/test_text_generator.py [hunk 31]
```diff
@@ -15,11 +15,19 @@ def test_return_type(self):
         for x in text_arr:
             self.assertIsInstance(x, np.str_)
 
-    def test_text_length(self):
+    def test_text_length_range(self):
         text_arr = random_text(self.rng, str_len_min=4, str_len_max=5)
         self.assertLessEqual(len(text_arr[0]), 5)
```
Comment (taylorfturner) [inline tests/distinct_generators/test_text_generator.py:20]:
```suggestion
        self.assertEquals(len(text_arr), 1)
        self.assertLessEqual(len(text_arr[0]), 5)
```

File: tests/test_generators.py [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+            {
+                "data_labeler.is_enabled": False,
+                "correlation.is_enabled": True,
+                "multiprocess.is_enabled": False,
+            }
+        )
+        dp.set_seed(0)
+
+        # create dataset and profile for tabular
+        cls.tab_csv_data = dp.Data(os.path.join(test_dir, "data/tabular.csv"))
```
Comment (taylorfturner) [inline tests/test_generators.py:29]:
```suggestion
        cls.data = dp.Data(os.path.join(test_dir, "data/tabular.csv"))
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: tests/distinct_generators/test_text_generator.py [hunk 33]
```diff
@@ -15,10 +15,17 @@ def test_return_type(self):
...
             self.assertIsInstance(x, np.str_)
 
-    def test_text_length(self):
-        text_arr = random_text(self.rng, str_len_min=4, str_len_max=5)
-        self.assertLessEqual(len(text_arr[0]), 5)
-        self.assertGreaterEqual(len(text_arr[0]), 4)
+    def test_text_length_range(self):
+        text_arr = random_text(self.rng, str_len_min=3, str_len_max=5)
+        for text in text_arr:
+            self.assertLessEqual(len(text), 5)
```
Comment (taylorfturner) [inline tests/distinct_generators/test_text_generator.py:21]:
```suggestion
            self.assertLess(len(text), 5)
```
if exclusive

File: tests/distinct_generators/test_text_generator.py [hunk 34]
```diff
@@ -15,10 +15,17 @@ def test_return_type(self):
...
+            self.assertGreaterEqual(len(text), 3)
+
+    def test_text_equal_length_range(self):
+        text_arr = random_text(self.rng, str_len_min=5, str_len_max=5)
+        for text in text_arr:
+            self.assertLessEqual(len(text), 5)
```
Comment (taylorfturner) [inline tests/distinct_generators/test_text_generator.py:27]:
```suggestion
            self.assertLess(len(text), 5)
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/distinct_generators/int_generator.py (hunk 16 above)
Comment (taylorfturner) [inline synthetic_data/distinct_generators/int_generator.py:24]: I'd recommend updating docstring too on line 16-17

File: synthetic_data/distinct_generators/text_generator.py [hunk 35]
```diff
@@ -39,6 +39,10 @@ def random_text(
         )
     text_list = []
 
+    # edge case
```
Comment (ksneab7) [inline synthetic_data/distinct_generators/text_generator.py:42]:
```suggestion
    # Correction when max == min length, generation is exclusive of max length
```

File: synthetic_data/distinct_generators/int_generator.py [hunk 36]
```diff
@@ -20,4 +20,7 @@ def random_integers(
 
     :return: np array of integers
     """
+    if min_value == max_value:
```
Comment (ksneab7) [inline synthetic_data/distinct_generators/int_generator.py:23]:
```suggestion
   # Correction when max == min value, generation is exclusive of max value
    if min_value == max_value:
```

File: synthetic_data/generators.py (hunk 30 above)
Comment (taylorfturner) [reply synthetic_data/generators.py:86]: yeah we definitely to abstract this more as I've thought about it

**Comment by taylorfturner** [comment]
I've decided we can definitely abstract this more #alwaysabstractmore

File: synthetic_data/generators.py [hunk 37]
```diff
@@ -47,20 +75,164 @@ def post_profile_processing_w_data(cls, data, profile):
...
+            return make_data_from_report(
+                report=self.profile.report(),
+                n_samples=num_samples,
+                noise_level=noise_level,
+                seed=self.tabular_generator_seed,
+            )
+        else:
+            return self.generate_uncorrelated_column_data(num_samples)
+
+    def generate_uncorrelated_column_data(self, num_samples):
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply synthetic_data/generators.py:97]: I think to, just like `make_data_from_report` this should be not in the this `TabularGenerator` class...

File: synthetic_data/distinct_generators/int_generator.py [hunk 38]
```diff
@@ -4,20 +4,22 @@
...
+    :type min: int, optional
+    :param max: the maximum integer that can be returned
+    :type max: int, optional
     :param num_rows: the number of rows in np array generated
     :type num_rows: int, optional
 
     :return: np array of integers
     """
-    return rng.integers(min_value, max_value, (num_rows,))
+    max += 1
```
Comment (taylorfturner) [inline synthetic_data/distinct_generators/int_generator.py:23]:
```suggestion
    # because `rng.integers` has an exclusive `max`, we need to ensure the max of the data is 
    # `n-1` the max parameter value. 
    max += 1
```

Comment (taylorfturner) [reply synthetic_data/distinct_generators/int_generator.py:23]: something to this effect

File: synthetic_data/distinct_generators/text_generator.py (hunk 35 above)
Comment (taylorfturner) [reply synthetic_data/distinct_generators/text_generator.py:43]: +1

File: synthetic_data/generators.py [hunk 39]
```diff
@@ -47,20 +75,164 @@ def post_profile_processing_w_data(cls, data, profile):
...
+        dataset = []
+        column_names = []
+        sorting_types = ["ascending", "descending"]
+
+        for col in columns:
+            col_ = copy.deepcopy(col)
+
+            generator_name = col_.get("data_type", None)
+            generator_func = self.gen_funcs.get(generator_name, None)
+            print(generator_name)
```
Comment (taylorfturner) [inline synthetic_data/generators.py:109]:
```suggestion
```

File: tests/distinct_generators/test_text_generator.py (hunk 33 above)
Comment (taylorfturner) [reply tests/distinct_generators/test_text_generator.py:21]: @drahc1R ?

File: tests/test_generators.py
Comment (taylorfturner) [inline tests/test_generators.py:1]: this file still has a ton of commented out code... needs clean up

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/distinct_generators/float_generator.py [hunk 40]
```diff
@@ -5,29 +5,29 @@
 
 def random_floats(
     rng: Generator,
-    min_value: int = -1e6,
-    max_value: int = 1e6,
-    sig_figs: int = 3,
+    min: int = -1e6,
+    max: int = 1e6,
```
Comment (taylorfturner) [inline synthetic_data/distinct_generators/float_generator.py:9]:
```suggestion
    min: int,
    max: int,
```

File: synthetic_data/generators.py (hunk 37 above)
[the remark this answers is not in the document]
Comment (taylorfturner) [reply synthetic_data/generators.py:97]: but we can keep in this class for now... just make it hidden with a leading `_`

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: tests/test_generators.py
Comment (drahc1R) [reply tests/test_generators.py:1]: will work on it

File: synthetic_data/distinct_generators/text_generator.py [hunk 41]
```diff
@@ -8,29 +8,30 @@
 
 def random_text(
     rng: Generator,
-    chars: Optional[List[str]] = None,
-    str_len_min: int = 1,
-    str_len_max: int = 1000,
+    vocab: Optional[List[str]],
+    min: int,
```
Comment (taylorfturner) [inline synthetic_data/distinct_generators/text_generator.py:12]: @ksneab7 and I talked and we decided to maintain defaults

**Review by taylorfturner — APPROVED** [review]
LGTM

File: tests/test_generators.py [hunk 42]
```diff
@@ -0,0 +1,598 @@
...
+                "max_value": 10,
+                "sig_figs": 3,
+            },
+        ]
+
+    # @staticmethod
+    # def setup_tabular_generator_mock(mock_generator):
+    #     mock_DataLabeler = mock_generator.return_value
+
+    # TEST PARAM_BUILD
```
Comment (taylorfturner) [inline tests/test_generators.py:140]: remove

File: tests/test_generators.py [hunk 43]
```diff
@@ -0,0 +1,598 @@
...
+#     def test_generate_dataset_with_valid_sorting_type(self, mock_warning):
+#         columns_to_gen = [
+#             {
+#                 "generator": "integer",
+#                 "name": "int",
+#                 "min_value": 4,
+#                 "max_value": 88,
+#                 "order": "ascending",
+#             }
+#         ]
```
Comment (taylorfturner) [inline tests/test_generators.py:446]: huh?

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

**Review by ksneab7 — APPROVED** [review]
